test/Fees.t.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import {Test, console2} from "forge-std/Test.sol";
+import {PDPFees} from "../src/Fees.sol";
+
+contract PDPFeesTest is Test {
+    uint256 constant epochs_per_day = 2880;
+
+    function computeRewardPerPeriod(uint64 filUsdPrice, int32 filUsdPriceExpo, uint256 rawSize) internal pure returns (uint256) {
+        uint256 rewardPerEpochPerByte;
+        if (filUsdPriceExpo >= 0) {
+            rewardPerEpochPerByte = (PDPFees.MONTHLY_TIB_STORAGE_REWARD_USD * PDPFees.FIL_TO_ATTO_FIL) /
+                (PDPFees.TIB_IN_BYTES * PDPFees.EPOCHS_PER_MONTH * filUsdPrice * (10 ** uint32(filUsdPriceExpo)));
+        } else {
+            rewardPerEpochPerByte = (PDPFees.MONTHLY_TIB_STORAGE_REWARD_USD * PDPFees.FIL_TO_ATTO_FIL * (10 ** uint32(-filUsdPriceExpo))) /
+                (PDPFees.TIB_IN_BYTES * PDPFees.EPOCHS_PER_MONTH * filUsdPrice);
+        }
+        uint256 rewardPerPeriod = rewardPerEpochPerByte * epochs_per_day * rawSize;
+        return rewardPerPeriod;
+    }
+
+    function testProofFeeWithGasFeeBoundZeroGasFee() public {
+        vm.expectRevert("failed to validate: estimated gas fee must be greater than 0");
+        PDPFees.proofFeeWithGasFeeBound(0, 5, 0, 1e18, epochs_per_day);
+    }
+
+    function testProofFeeWithGasFeeBoundZeroAttoFilUsdPrice() public {
+        vm.expectRevert("failed to validate: AttoFIL price must be greater than 0");
+        PDPFees.proofFeeWithGasFeeBound(1, 0, 0, 1e18, epochs_per_day);
+    }
+
+    function testProofFeeWithGasFeeBoundZeroRawSize() public {
+        vm.expectRevert("failed to validate: raw size must be greater than 0");
+        PDPFees.proofFeeWithGasFeeBound(1, 5, 0, 0, epochs_per_day);
+    }
+
+    function testProofFeeWithGasFeeBoundHighGasFee() public pure {
+        uint64 filUsdPrice = 5;
+        int32 filUsdPriceExpo = 0;
+        uint256 rawSize = 1e18;
+
+        uint256 rewardPerPeriod = computeRewardPerPeriod(filUsdPrice, filUsdPriceExpo, rawSize);
+
+        uint256 gasLimitRight = (rewardPerPeriod * PDPFees.GAS_LIMIT_RIGHT_PERCENTAGE) / 100;
+
+        uint256 estimatedGasFee = gasLimitRight;
+
+        uint256 fee = PDPFees.proofFeeWithGasFeeBound(estimatedGasFee, filUsdPrice, filUsdPriceExpo, rawSize, epochs_per_day);
+
+        assertEq(fee, 0, "Fee should be 0 when gas fee is high");
+    }
+
+    function testProofFeeWithGasFeeBoundMediumGasFee() public pure {
+        uint64 filUsdPrice = 5;
+        int32 filUsdPriceExpo = 0;
+        uint256 rawSize = 1e18;
+
+        uint256 rewardPerPeriod = computeRewardPerPeriod(filUsdPrice, filUsdPriceExpo, rawSize);
+
+        uint256 gasLimitLeft = (rewardPerPeriod * PDPFees.GAS_LIMIT_LEFT_PERCENTAGE) / 100;
+        uint256 gasLimitRight = (rewardPerPeriod * PDPFees.GAS_LIMIT_RIGHT_PERCENTAGE) / 100;
+
+        uint256 estimatedGasFee = (gasLimitLeft + gasLimitRight) / 2;
```

Comment:
nit: would be nice to add to this case tests at the boundaries, i.e. estimatedGasFee = gasLimitLeft and estimatedGasFee = gasLimitRight - 1

Reply:
Done.